packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +71,247 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const path = globalThis.window.location.pathname;
+
+      let pattern = argument;
+      if (pattern.startsWith('/') && pattern.endsWith('/')) {
+        pattern = pattern.slice(1, -1);
+      }
+
+      const regex = new RegExp(pattern);
+      return regex.test(path);
+    } else if (selector.name === 'matches-attr') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const indexOfEqual = argument.indexOf('=');
+      let namePattern, valuePattern;
```

Comment:
I think we need to handle cases where the name is quoted as well here.

> A declaration in the form name="value" or "name"="value",

From: https://github.com/gorhill/uBlock/wiki/Procedural-cosmetic-filters#subjectmatches-attrarg